### PR TITLE
Create user account endpoint

### PR DESCRIPTION
**Pull request by Rexben001**
# Description

An endpoint to create a user account(Staff)

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const { NODE_ENV } = process.env;
+let connectionString = '';
+
+if (NODE_ENV === 'test') {
+  connectionString = process.env.DB_CONNECTION_TEST;
+} else {
+  connectionString = process.env.DB_CONNECTION;
+}
+
+module.exports = connectionString;
```
Comment (Pav0l) [inline config/config.js]: Great use of `config` file 👍

File: controllers/UserController.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  static async createUser(req, res) {
+    try {
+      const password = await hashPassword(req.body.password);
+      req.body.password = password;
+      const result = await models.User.create(req.body);
+      return res.status(201).json({ result });
+    } catch (err) {
+      if (err.code === 11000) {
+        return res.status(409).json({ message: 'username exists already' });
+      }
```
Comment (Pav0l) [inline controllers/UserController.js]: I'm worried this error handling is scalable. As there are many `err.code`s which we will not be able to cover. Maybe we can just return the `err.message`?

File: package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "node start.js",
     "dev": "NODE_ENV=development nodemon start.js",
-    "test": "mocha --timeout 10000 tests/*.js --exit"
+    "test": "NODE_ENV=test mocha --timeout 10000 tests/*.js --exit"
```
Comment (Pav0l) [inline package.json]: ❓ Will this work without `cross env`?

**Comment by Pav0l** [comment]
[Create API endpoints for User](https://trello.com/c/X2KZIBxp/23-create-api-endpoints-for-user)

File: start.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 const app = require('./index');
+const config = require('./config/config');
```
[the remark this answers is not in the document]
Comment (Rexben001) [reply start.js]: I forgot to clean up my code

File: package.json (hunk 3 above)
Comment (Rexben001) [reply package.json]: Yes but to be on a safer side, I'd use `cross-env`

**Pull request merged**